twa

```diff
@@ -118,31 +122,50 @@ function stage_0_has_https {
   # First, just probe the domain to see if it listens on 443.
   if ! nc -z -w2 "${domain}" 443 2>/dev/null; then
     FATAL "expected port 443 to be open, but it isn't"
-    no_https=1 ; return
+    no_https=1
   fi
 
-  # The rest below would be cool, but is probably unnecessary since `curl`
-  # won't connect to an insecure HTTPS server by default.
-  # # Then, use `openssl` to retrieve the certificate and test its validity.
-  # cert=$(openssl s_client -connect "${domain}:443" < /dev/null)
+  fetch_secure -o /dev/null "https://${domain}" 2>/dev/null
+  if [ "${?}" -eq 0 ]; then
```

Comment:
Just as a general style suggestion: you can test return commands directly via `if`:

```bash
if fetch "https://${domain}"; then
  echo "command succeeded"
fi
```